From: Alexandre Belloni <alexandre.belloni@bootlin.com>
Date: Wed, 3 Apr 2019 16:11:19 +0200
Subject: [PATCH 11/12] misc: atmel_tclib: do not probe already used TCBs

The TCBs that have children are using the new (proper) DT bindings and
don't need to be handled by tclib.

Signed-off-by: Alexandre Belloni <alexandre.belloni@bootlin.com>
Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 drivers/misc/atmel_tclib.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/misc/atmel_tclib.c
+++ b/drivers/misc/atmel_tclib.c
@@ -79,6 +79,9 @@ static int __init tc_probe(struct platfo
 	struct resource	*r;
 	unsigned int	i;
 
+	if (of_get_child_count(pdev->dev.of_node))
+		return 0;
+
 	irq = platform_get_irq(pdev, 0);
 	if (irq < 0)
 		return -EINVAL;
